### PR TITLE
Ruby: Add rb/insecure-dependency query

### PR DESCRIPTION
**Pull request by hmac**
This is a port of the JS query of the same name. It looks for places in a `Gemfile` where a `http` or `ftp` URL is specified. The tests are written using inline expectations.

### CODE REVIEW

**Review by nickrolfe — COMMENTED** [review]
Very nice. I have just a few small comments.

File: ruby/ql/src/queries/security/cwe-300/InsecureDependencyResolution.ql [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * @id rb/insecure-dependency
+ * @tags security
+ *       external/cwe/cwe-300
+ *       external/cwe/cwe-319
+ *       external/cwe/cwe-494
+ *       external/cwe/cwe-829
+ */
+
+import ruby
+import codeql.ruby.security.InsecureDependencyQuery
```
[the remark this answers is not in the document]
Comment (hmac) [reply ruby/ql/src/queries/security/cwe-300/InsecureDependencyResolution.ql:18]: Yeah exactly - the test can only see files in `lib`, AFAIK.

File: ruby/ql/lib/codeql/ruby/security/InsecureDependencyQuery.qll [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  proto = s.regexpCapture("^(http|ftp):.+", 1).toUpperCase()
+}
+
+/**
+ * Holds if `e` is a string containing a URL that uses the insecure protocol `proto`.
+ */
+private predicate containsInsecureUrl(Expr e, string proto) {
+  // Handle cases where the string as a whole has no constant value (due to interpolations)
+  // but has a known prefix. E.g. "http://#{foo}"
+  exists(StringComponent c | c = e.(StringlikeLiteral).getComponent(_) |
```
[the remark this answers is not in the document]
Comment (hmac) [reply ruby/ql/lib/codeql/ruby/security/InsecureDependencyQuery.qll:62]: Ah good catch! Yes.

**Review by nickrolfe — DISMISSED** [review]
🚀

**Review by nickrolfe — APPROVED** [review]

**Pull request merged**